server/src/WifiAccessPointLocalConfig.c

```diff
@@ -0,0 +1,244 @@
+#include <WifiAccessPointLocalConfig.h>
+#include <ctype.h>
+
+void print_element_names(xmlNode * a_node);
+
+static xmlNode * findNode(const char *path) //find xmlnode in memory corresping to arg path
+{
+   xmlNode *curr = firstRootChild;   //defined during lcfg_init as the first rootnode child
+
+   char *pathName = strstr(path,"/WifiNodes/");  //needle-hay search for /WifiNodes/ in path. pathName points to first position of path that corresponds to needle, /WifiNodes/
+
+   if(pathName == NULL) // check to make sure path provided is valid per system's xml
+   {
+      return NULL;
+   }
+
+   //print_element_names(curr->parent);
+
+   pathName += strlen("/WifiNodes/");
+
+   while((curr != NULL)  && (pathName != NULL))
+   {
+
+      curr = xmlNextElementSibling(curr);
+
+      if (curr)
+      {
+         size_t len = strlen((const char *)curr->name);
+         if(!strncmp(pathName,(const char*)curr->name,len))
+         {
+            curr = curr->children;
+            pathName = strstr((const char*)pathName,"/");
+            if (pathName)
+            {
+               //printf("match: %s\n",pathName);
+               pathName++;
+            }
+         }
+      }
+   }
+   return curr;
+}
+
+void print_element_names(xmlNode * a_node)
+{
+    xmlNode *cur_node = NULL;
+
+    for (cur_node = a_node; cur_node; cur_node = cur_node->next) {
+        if (cur_node->type == XML_ELEMENT_NODE) {
+            printf("node type: Element, name: %s\n", cur_node->name);
+        }
+
+        print_element_names(cur_node->children);

```

Comment:
Is this a recursion?! Cool, but isn't it dangerous to do such in C?

Also, why do we need this `print_element_names` function if no one is using it?

Reply:
Yeh, it is not used. Iterating through all the leaves of a tree can be a daunting task. I only included it for the future. But as for recursion being dangerous, it could be because of infinite recursion possibilities. But in this case, there is a for loop which terminates when a cur_node is null. That makes it amicable.